engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/transform/FormResourceTransformer.java

```diff
@@ -0,0 +1,152 @@
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.deployment.transform;
+
+import static io.camunda.zeebe.util.buffer.BufferUtil.wrapString;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.zeebe.engine.processing.common.Failure;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.StateWriter;
+import io.camunda.zeebe.engine.state.immutable.FormState;
+import io.camunda.zeebe.protocol.impl.record.value.deployment.DeploymentRecord;
+import io.camunda.zeebe.protocol.impl.record.value.deployment.DeploymentResource;
+import io.camunda.zeebe.protocol.impl.record.value.deployment.FormMetadataRecord;
+import io.camunda.zeebe.protocol.impl.record.value.deployment.FormRecord;
+import io.camunda.zeebe.protocol.record.intent.FormIntent;
+import io.camunda.zeebe.stream.api.state.KeyGenerator;
+import io.camunda.zeebe.util.Either;
+import java.io.IOException;
+import java.util.function.Function;
+import java.util.function.LongSupplier;
+import org.agrona.DirectBuffer;
+
+public final class FormResourceTransformer implements DeploymentResourceTransformer {
+
+  private static final int INITIAL_VERSION = 1;
+
+  private static final Either<Failure, Object> NO_DUPLICATES = Either.right(null);
+
+  private static final ObjectMapper JSON_MAPPER = new ObjectMapper();
+
+  private final KeyGenerator keyGenerator;
+  private final StateWriter stateWriter;
+  private final Function<DeploymentResource, DirectBuffer> checksumGenerator;
+  private final FormState formState;
+
+  public FormResourceTransformer(
+      final KeyGenerator keyGenerator,
+      final StateWriter stateWriter,
+      final Function<DeploymentResource, DirectBuffer> checksumGenerator,
+      final FormState formState) {
+    this.keyGenerator = keyGenerator;
+    this.stateWriter = stateWriter;
+    this.checksumGenerator = checksumGenerator;
+    this.formState = formState;
+  }
+
+  @Override
+  public Either<Failure, Void> transformResource(
+      final DeploymentResource resource, final DeploymentRecord deployment) {
+
+    return parseFormId(resource)
+        .map(
+            formId -> {
+              final FormMetadataRecord formRecord = deployment.formMetadata().add();
+              appendMetadataToFormRecord(formRecord, formId, resource);
+              writeFormRecord(formRecord, resource);
+
+              return null;
+            });
+  }
+
+  private Either<Failure, String> parseFormId(final DeploymentResource resource) {
+    try {
+      final String formId = JSON_MAPPER.readValue(resource.getResource(), FormIdPOJO.class).getId();
```

Comment:
@nicpuppa as I observed from the codebase, we use `ObjectMapper` in other classes to parse json. Why do you think we shouldn't use `ObjectMapper` here?

Reply:
`ObjectMapper` is used only in exporters or in the java client. Idk if we want to use `ObjectMapper` inside the engine.

Reply:
Since it was not needed so far in the `engine`, it was not used. But when Json parsing is needed e.g. in exporters and client `jackson` is used. It is also defined in the parent `pom.xml`. It should be the correct way to go. Let me also bring it to the team, so we make sure.